WoWDatabaseEditor.Common/WDE.CMMySqlDatabase/Database/BaseDatabaseProvider.cs

```diff
@@ -0,0 +1,473 @@
+        }
+
+        public async Task<List<IAreaTriggerTemplate>> GetAreaTriggerTemplatesAsync()
+        {
+            if (currentCoreVersion.Current.DatabaseFeatures.UnsupportedTables.Contains(typeof(IAreaTriggerTemplate)))
+                return new List<IAreaTriggerTemplate>();
+
+            await using var model = Database();
+            return await (from t in model.AreaTriggerTemplate orderby t.Id select t).ToListAsync<IAreaTriggerTemplate>();
+        }
+        
+        public IEnumerable<IGameObjectTemplate> GetGameObjectTemplates()
+        {
+            using var model = Database();
+            return (from t in model.GameObjectTemplate orderby t.Entry select t).ToList<IGameObjectTemplate>();
+        }
+        
+        public async Task<List<IGameObjectTemplate>> GetGameObjectTemplatesAsync()
+        {
+            await using var model = Database();
+            var o = from t in model.GameObjectTemplate orderby t.Entry select t;
+            return await (o).ToListAsync<IGameObjectTemplate>();
+        }
+
+        public abstract Task<List<IGameObject>> GetGameObjectsAsync();
+
+        protected virtual IQueryable<QuestTemplateWoTLK> GetQuestsQuery(BaseDatabaseTables model)
+        {
+            return from t in model.QuestTemplate
+                orderby t.Entry
+                select t;
+        }
+        
+        public virtual IEnumerable<IQuestTemplate> GetQuestTemplates()
+        {
+            using var model = Database();
+
+            return GetQuestsQuery(model).ToList<IQuestTemplate>();
+        }
+
+        public virtual async Task<List<IQuestTemplate>> GetQuestTemplatesAsync()
+        {
+            await using var model = Database();
+            var o = GetQuestsQuery(model);
+            return await o.ToListAsync<IQuestTemplate>();
+        }
+
+        public virtual IQuestTemplate? GetQuestTemplate(uint entry)
+        {
+            using var model = Database();
+            return model.QuestTemplate.FirstOrDefault(q => q.Entry == entry);
+        }
+
+        public IGameObjectTemplate? GetGameObjectTemplate(uint entry)
+        {
+            using var model = Database();
+            return model.GameObjectTemplate.FirstOrDefault(g => g.Entry == entry);
+        }
+
+        public async Task InstallScriptFor(int entryOrGuid, SmartScriptType type, IList<ISmartScriptLine> script)
+        {
+        }
+
+        protected abstract Task<ICreature?> GetCreatureByGuid(T model, uint guid);
+        protected abstract Task<IGameObject?> GetGameObjectByGuidAsync(T model, uint guid);
+        protected abstract Task SetCreatureTemplateAI(T model, uint entry, string ainame, string scriptname);
+
+        public async Task InstallConditions(IEnumerable<IConditionLine> conditionLines,
+            IDatabaseProvider.ConditionKeyMask keyMask,
+            IDatabaseProvider.ConditionKey? manualKey = null)
+        {
+            using var writeLock = await DatabaseLock.WriteLock();
```

Comment:
Conditions in cmangos are also totally different, right?  So in the future it will be easier to just add another cmangos specific interfaces and methods for its conditions, and this method should do nothing, or maybe even throw an exception.

Reply:
code removed 👍